File: 08-2-2-Adapter-Solution/Ducks/TurkeyAdapter.cs
namespace Adapter.Ducks
{
    using System;

    using Adapter.Fly;
    using Adapter.Quack;
    using Adapter.Turkeys;

    public class TurkeyAdapter : Duck
    {
        private readonly Turkey _turkey;

        public TurkeyAdapter(Turkey turkey) : base(new TurkeyQuackStrategy(turkey), new TurkeyFlyStrategy(turkey))
        {
            _turkey = turkey;
        }

        public override void Swim() => _turkey.Swim();

        public override void Display() => Console.WriteLine("I am a Turkey hiding at the Duck Pond.");
    }
}

File: 08-DesignPatterns-Java/Solution/Adapter/src/Ducks/TurkeyAdapter.java
package Ducks;

import Fly.TurkeyFlyStrategy;
import Quack.TurkeyQuackStrategy;

public class TurkeyAdapter extends Duck {
    private final Turkeys.Turkey _turkey;

    public TurkeyAdapter(Turkeys.Turkey turkey) {
        super(new TurkeyQuackStrategy(turkey), new TurkeyFlyStrategy(turkey));
        _turkey = turkey;
    }

    @Override
    public void swim() {
        _turkey.swim();
    }

    @Override
    public void display() {
        System.out.println("I am a Turkey hiding at the Duck Pond.");
    }
}
